### PR TITLE
refactor lookups to be more chill to router

### PR DESCRIPTION
**Pull request by clintropolis**
This PR introduces `LookupSerdeModule` which allows the router to perform query serde but without having to load lookups as happens using the stock `LookupModule` with default settings. To achieve this, an interface has been extracted from `LookupReferencesManager`, the very java named `LookupExtractorFactoryContainerProvider`, which is now used everywhere `LookupReferencesManager` was previously used, _except_ for `LookupListeningResource`. I think this is a bit nicer, because now all query processing stuff no longer has obvious access to methods to load and drop lookups, only the method to get the `LookupExtractorFactoryContainer` needed to do the transforms. Additionally, this has allowed many of these types to be migrated into `druid-processing` which seems like a more appropriate home for them (including the new `LookupExtractorFactoryContainerProvider` interface).

Now in `druid-processing`:
* `LookupDimensionSpec`
* `LookupExprMacro`
* `LookupExtractorFactoryContainerProvider`
* `RegisteredLookupExtractionFn`

An integration test has been added to the wiki tests, which loads a simple json lookup, and tests queries against brokers and routers. This also means there is now an integration test that is testing lookups! I've also tested this on a small test cluster spun up on my laptop.

The router will proxy requests to the lookup introspection API it was previously serving, so if the router was being queried to introspect lookups things should still function, it will just be proxied to a default broker now.

I suspect it wasn't _too much_ of an issue, because that API didn't appear functional because didn't have auth checks performed, and would throw an exception of the form
```
2019-03-10T00:15:20,298 WARN [qtp2080864181-118] org.eclipse.jetty.server.HttpChannel - /druid/v1/lookups/introspect/simple-wiki
org.apache.druid.java.util.common.ISE: Request did not have an authorization check performed.
    at org.apache.druid.server.security.PreResponseAuthorizationCheckFilter.handleAuthorizationCheckError(PreResponseAuthorizationCheckFilter.java:156) ~[classes/:?]
    at org.apache.druid.server.security.PreResponseAuthorizationCheckFilter.doFilter(PreResponseAuthorizationCheckFilter.java:89) ~[classes/:?]
...
```
So I've included a fix for that as well.

I also made some minor adjustments to the integration tests configurations, switching gc from cms to g1, and lowering some thread pool sizes to maybe result in a slightly lighter footprint.

### CODE REVIEW

**Comment by clintropolis** [comment]
Oops, test failures or legit, fixing the `LookupIntrospectionResource` auth check error has broken `LookupIntrospectionResourceImplTest`, will fix.

**Comment by clintropolis** [comment]
Fixed and combined both `LookupIntrospectionResource` tests, reworking `LookupIntrospectionResourceImplTest` by [borrowing some ideas from past me](https://github.com/apache/incubator-druid/pull/6799/files#diff-20e188fb124810afa111d1ccaccf5f84R95) and putting into a utility class, `WebserverTestUtils`, to help make a test webserver that works with both jersey and guice injections in order to continue test API function like `LookupIntrospectionResourceImplTest` was doing. Hopefully this should make writing other tests of this manner a bit easier than requiring full integration tests.

**Comment by clintropolis** [comment]
Would the module introduced in this PR also be a good solution to the problem of #5727 since it would allow lookup serde without loading lookups? Perhaps I could rename `RouterLookupModule` to something more generic if so.

**Comment by clintropolis** [comment]
>Perhaps I could rename RouterLookupModule to something more generic if so.

Renamed to `LookupSerdeModule`

**Comment by yurmix** [comment]
This is interesting.
@drcrallen, is this the interface extraction you were referring to in [Re: Druid module separation (And a related lookups question)](https://lists.apache.org/thread.html/a26216e04012b50fa3a201ddd4e8e5daac30b8bbad95740ab6a2d780@%3Cdev.druid.apache.org%3E)?
I wonder this pr now makes it possible to migrate `LookupDimensionSpec` to `druid-processing`. It will help me with the implementation of [multi-value lookups](https://github.com/apache/incubator-druid/issues/7009).

**Comment by clintropolis** [comment]
>This is interesting.
@drcrallen, is this the interface extraction you were referring to in Re: Druid module separation (And a related lookups question)?

Ah, sorry I hadn't got around to that dev list thread yet, I'll have a look and see if I can move some stuff around since I have to fix up conflicts anyway.

**Comment by clintropolis** [comment]
@yurmix @drcrallen I was able to refactor to move these things to the `druid-processing` module since most lookup things just refer to the `LookupExtractorFactoryContainerProvider` interface 
introduced in this PR:
* `LookupDimensionSpec`
* `LookupExprMacro`
* `LookupExtractorFactoryContainerProvider`
* `RegisteredLookupExtractionFn`

**Comment by yurmix** [comment]
Wow, thanks so much! Now I'm going to reinitiate my efforts on multi-value lookups.

File: services/src/main/java/org/apache/druid/cli/CliRouter.java [hunk 1]
```diff
@@ -80,44 +79,39 @@ public CliRouter()
...
         JettyHttpClientModule.global(),
-        new Module()
-        {
-          @Override
-          public void configure(Binder binder)
-          {
-            binder.bindConstant().annotatedWith(Names.named("serviceName")).to("druid/router");
-            binder.bindConstant().annotatedWith(Names.named("servicePort")).to(8888);
-            binder.bindConstant().annotatedWith(Names.named("tlsServicePort")).to(9088);
+        binder -> {
```
Comment (leventov) [inline services/src/main/java/org/apache/druid/cli/CliRouter.java]: I'm not sure an anonymous class conversion to a lambda makes code better in this case. You lose visibility into the fact that this lambda is a `Module` and that the method is `configure()`, making the code more cryptic, IMO.

**Pull request merged**